force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +234,13 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
-        // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        // Use an if statement to check if d1 or d2 are null. If so, return null.
+        if (d1 != null && d2 != null) {
+            // Subtract the difference in years between Date 2 and Date 1 and return the result
+            Integer diffInYears = d2.year() - d1.year();
+            return diffInYears;
+        }
+        return null;
```

Comment:
Good solution. From a style point of view I like to leave what I think of that main code path out of the conditional. And my conditional only handles alternate scenarios. 

```
 public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return d2.year() - d1.year();
        // Subtract the difference in years
    }
 ```